src/Tests/Microsoft.DotNet.GenAPI.Tests/CSharpFileBuilderTests.cs

```diff
@@ -1116,5 +1116,42 @@ public partial class C : B
                     }
                     """);
         }
+
+        [Fact]
+        public void TestBaseTypeConstructorWithObsoleteAttribute()
+        {
+            RunTest(original: """
+                    namespace Foo
+                    {
+                        public class B
+                        {
+                            public B(int p1, string p2) { }
+                            [System.Obsolete("Constructor is deprecated.", true)]
```

Comment:
Can you please add two more test case for when the obsoleteattribute is applied but without the error: true parameter and when the obsolete message is null?

Reply:
Added 2 more unit tests